src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -72,7 +72,7 @@ pub fn caml_pasta_fp_plonk_proof_create(
             .collect()
     };
 
-    let witness: Vec<Vec<_>> = witness.iter().map(|x| (*x.0).clone()).collect();
+    let witness: Vec<Vec<_>> = witness.iter().map(|x| (**x).clone()).collect();
```

Comment:
This is calling Deref on the `Rc`, which gives the `UnsafeCell` value, which is, in memory, the same as the underlying type `T` (see [this comment](https://github.com/MinaProtocol/mina/pull/16648/files#r1970498062)).